tests/acceptance/async_test_base.py

```diff
@@ -91,11 +92,10 @@ def on_open_error(self, connection, error):
 
     def on_timeout(self):
         """called when stuck waiting for connection to close"""
-        self.logger.info('on_timeout at %s', datetime.utcnow())
-        # force the ioloop to stop
-        self.logger.debug('on_timeout called')
-        self.connection.ioloop.stop()
-        raise AssertionError('Test timed out')
+        self.logger.info('on_timeout called at %s', datetime.utcnow())
+        # Initiate cleanup
+        self.timeout = None  # the dispatcher should have removed it
+        self.stop()
```

Comment:
@gst, please be careful to preserve these changes in `on_timeout`, too, when merging #678 that modifies the same method. I asked @gmr to merge this PR #683 ahead of the other PRs so that the other PRs may be re-merged with master after #683 to verify clean builds in all of them.